clipboard/src/web.ts

```diff
@@ -0,0 +1,100 @@
+import { WebPlugin } from '@capacitor/core';
+import {
+  ClipboardPlugin,
+  ClipboardWrite,
+  ClipboardReadResult,
+} from './definitions';
+
+declare var navigator: any;
+declare var ClipboardItem: any;
+
+export class ClipboardWeb extends WebPlugin implements ClipboardPlugin {
+  constructor() {
+    super({
+      name: 'Clipboard',
+      platforms: ['web'],
+    });
+  }
+
+  async write(options: ClipboardWrite): Promise<void> {
+    if (!navigator.clipboard) {
+      return Promise.reject('Clipboard API not available in this browser');
+    }
+
+    if (options.string !== undefined || options.url) {
+      if (!navigator.clipboard.writeText) {
+        return Promise.reject(
+          'Writting to clipboard not supported in this browser',
+        );
+      }
+      await navigator.clipboard.writeText(
+        options.string !== undefined ? options.string : options.url,
+      );
+    } else if (options.image) {
+      if (!navigator.clipboard.write) {
+        return Promise.reject('Setting images not supported in this browser');
+      }
+      try {
+        const blob = await (await fetch(options.image)).blob();
+        const clipboardItemInput = new ClipboardItem({ [blob.type]: blob });
+        await navigator.clipboard.write([clipboardItemInput]);
+      } catch (err) {
+        return Promise.reject('Failed to write image');
+      }
+    } else {
+      return Promise.reject('Nothing to write');
+    }
+    return Promise.resolve();
+  }
+
+  async read(): Promise<ClipboardReadResult> {
+    if (!navigator.clipboard) {
+      return Promise.reject('Clipboard API not available in this browser');
+    }
+    if (!navigator.clipboard.read) {
+      if (!navigator.clipboard.readText) {
+        return Promise.reject(
+          'Reading from clipboard not supported in this browser',
+        );
+      }
+      return this.readText();
+    } else {
+      try {
+        const clipboardItems = await navigator.clipboard.read();
+        const type = clipboardItems[0].types[0];
+        const clipboardBlob = await clipboardItems[0].getType(type);
+        const data = await this._getBlobData(clipboardBlob, type);
+        return Promise.resolve({ value: data, type });
```

Comment:
Not necessary in `async` functions.

```suggestion
        return { value: data, type };
```